woocommerce/woocommerce-gutenberg-products-block
# Add hook documentation and build script

This PR takes care of:

- A new build script to generate hook documentation for actions and filters
   - This also supports an `@examples` tag for linking to an MD file with further code examples
   - Docs are added inside docs/extensibility/filters.md and docs/extensibility/actions.md
   - TOC generation is automatic
- Inline documentation for all filters and actions in our plugin on the PHP side
- Doc examples for some of the experimental filters where needed

![Screenshot 2021-10-12 at 17 13 59](https://user-images.githubusercontent.com/90977/136992440-7a03f803-3424-4ac9-b404-407fc255b928.png)

Updating docs consists of running the following command:

```bash
npm run build:docs
```

Why do this? Because duplicating content between docs and inline is time-consuming and can become out of sync. Having a script can automate this and allows us to document in one place (inline).

Closes #4629

### Testing

1. Composer install dependencies
2. Npm install dependencies
3. Delete docs/extensibility/filters.md and docs/extensibility/actions.md
4. Run `npm run build:docs`. See that docs were generated.
